gap/PolygonalComplexes/graphs.gi

```diff
@@ -1115,4 +1115,319 @@ InstallMethod(AllSimplicialSurfacesOfDigraph,"for a digraph and a Boolean",
+            return ErrorNoReturn("The given edge is not a digraph edge of the given digraph");
+        fi;
+
+        return EdgeInsertionNC(D, edgeA, edgeB);
+    end
+);
+
+BindGlobal( "__SIMPLICIAL_EdgeReductionDirectNC",
+    function (D, intersectingEdge)
+        local neighbours, neighbour, newEdge;
+
+        # Add new edges
+        #
+        # left side of intersecting edge
+        newEdge := [];
+        neighbours := OutNeighboursOfVertex(D, intersectingEdge[1]);
+        for neighbour in neighbours do
+            if neighbour <> intersectingEdge[2] then
+                Add(newEdge, neighbour);
+            fi;
+        od;
+        for neighbour in neighbours do
+            DigraphRemoveEdge(D, [intersectingEdge[1], neighbour]);
+        od;
+        __SIMPLICIAL_DigraphAddEdgeNC(D, newEdge[1], newEdge[2]);
+        __SIMPLICIAL_DigraphAddEdgeNC(D, newEdge[2], newEdge[1]);
+        #
+        # right side of intersecting edge
+        newEdge := [];
+        neighbours := OutNeighboursOfVertex(D, intersectingEdge[2]);
+        for neighbour in neighbours do
+            if neighbour <> intersectingEdge[1] then
+                Add(newEdge, neighbour);
+            fi;
+        od;
+        for neighbour in neighbours do
+            DigraphRemoveEdge(D, [intersectingEdge[2], neighbour]);
+        od;
+        __SIMPLICIAL_DigraphAddEdgeNC(D, newEdge[1], newEdge[2]);
+        __SIMPLICIAL_DigraphAddEdgeNC(D, newEdge[2], newEdge[1]);
+
+        # Remove old vertices
+        DigraphRemoveVertices(D, intersectingEdge);
+
+        return D;
+    end
+);
+
+InstallMethod( EdgeReductionNC, "for a digraph and a list", [IsDigraph, IsList],
+    function (D, intersectingEdge)
+        local dMutable, isMutable;
+
+        if not IsSymmetricDigraph(D) then
+            D := DigraphSymmetricClosure(D);
+        fi;
+
+        if IsMutableDigraph(D) then
+            isMutable := true;
+            dMutable := D;
+        else
+            isMutable := false;
+            dMutable := DigraphMutableCopy(D);
+        fi;
+
+        dMutable := __SIMPLICIAL_EdgeReductionDirectNC(dMutable, intersectingEdge);
+
+        if isMutable then
+            return dMutable;
+        fi;
+        return MakeImmutable(dMutable);
+    end
+);
+
+InstallMethod( EdgeReduction, "for a digraph and a list", [IsDigraph, IsList],
+    function (D, edge)
+        local edgeOutNeighbours;
+
+        if not IsSymmetricDigraph(D) then
+            D := DigraphSymmetricClosure(D);
+        fi;
+
+        if not IsDigraphEdge(D, edge) then
+            return ErrorNoReturn("The given edge is not a digraph edge of given digraph");
+        fi;
+
+        # Ensure edge has 3 or 4 out neighbours
+        if Length(OutNeighboursOfVertex(D, edge[1])) <> 3 or
+           Length(OutNeighboursOfVertex(D, edge[2])) <> 3 then
+            return ErrorNoReturn("The vertices of the given edge must have three vertices");
```

Comment:
This error message doesn't make sense, maybe: "The vertices of the given edge must have three neighbours" and the comment in line 1302 doesn't match. What happens if the edge is part of a triangle? Did we talked about this?